File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/rotativaService.js
import api from "./pedidosAPI";

class RotativaService {
  buscarTodas() {
    return api.get("/rotativas");
  }

  buscarPorId(id) {
    return api.get(`/rotativas/${id}`);
  }

  crear(data) {
    return api.post("/rotativas", data);
  }

  modificar(id, data) {
    return api.put(`/rotativas/${id}`, data);
  }

  eliminar(id) {
    return api.delete(`/rotativas/${id}`);
  }

  buscarPorDescripcion(descripcion) {
    return api.get(`/rotativas?descripcion=${descripcion}`);
  }
}

const service = new RotativaService();
export default service 

File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/RotativaService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Rotativa;

public interface RotativaService {
    public Rotativa crear(Rotativa rotativa);
	public Rotativa modificar(Rotativa rotativa);
	public void eliminar(Rotativa rotativa);
	
	public List<Rotativa> buscarTodos();
	public List<Rotativa> buscarPorNombre(String nombre);
	public Optional<Rotativa> buscarPorId(long  id);
    public List<Rotativa> buscarPorDescripcion(String patron);
}
